vignettes/simulations.Rmd

```diff
@@ -0,0 +1,860 @@
+
+```{r, parameters-function}
+# simulation parameters using truncated normal distribution
+simulate_parameters_trunc <-
+  
+  function(n, lower, upper, mean, sd, number) {
+    simulate <- sapply(1:1000, function(i)
+      rtruncnorm(
+        n = n,
+        a = lower,
+        b = upper,
+        mean = mean,
+        sd = sd
+      )) |>
+      as.data.frame()
+    
+    simulate_long <- simulate |>
+      pivot_longer(cols = everything(),
+                   names_to = "Simulation",
+                   values_to = "Value") |>
+      arrange(Simulation)
+    
+    result <-
+      simulate_long |>
+      group_by(Simulation) |>
+      transmute(Simulation, n = sum(Value <= number)) |>
+      distinct()
+    
+    mean <- mean(result$n)
+    
+    return(mean)
+}
+```
+
+```{r, parameters-result}
+# Using the function for covariates
+
+# hba1c
+hba1c =
+  simulate_parameters_trunc(46, 0, 11, 7.41, 1.33, 9)
+
+# tpo2
+tpo2 =
+  simulate_parameters_trunc(46, 30, 100, 53.4, 18.4, 50)
+
+# age
+age =
+  simulate_parameters_trunc(46, 0, 100, 59.2, 9.7, 55)
+
+# wound_size
+wound_size =
+  simulate_parameters_trunc(46, 0, 20, 2.7, 2.28, 2)
+```
+
+The results are presented in a table, assuming that the outcome refers to the first category of each parameter.
+
+```{r, parameters-result-table, tab.cap = "Summary of literature verification about strata selected parameters (Mrozikiewicz-Rakowska et. al., 2023)"}
+# summary table
+data.frame(hba1c, tpo2, age, wound_size) %>%
+  rename('wound size' = wound_size) %>%
+  pivot_longer(cols = everything(),
+               names_to = "parametr",
+               values_to = "n") %>%
+  mutate(
+    n = ceiling(n),
+    percent = round(n / 46, 2),
+    strata = c('<=9', '<=50', '<=55', '<=2')
+  ) %>%
+  rename('Percent of N = 46' = percent) %>%
+  gt()
+```
+
+## Generate data using Monte-Carlo simulations
+
+To generate data based on the assumptions summarized in the last section and taking into account the suggestions from physicians, a function called `build_prior_df` was created. This function, relying on the conjugate beta distribution, generated random samples based on the defined parameters for various variables. Conjugate priors have the advantageous property that, after incorporating data (likelihood), the posterior distribution is of the same type as the prior, simplifying calculations and result interpretation (@hodel2023beta).
+
+```{r, build-prior}
+set.seed(123)
+
+# create beta distribution parameters
+build_prior_df <- function(no_of_iterations, n) {
+  binary_matrix <- tribble(
+    ~ var,
+    ~ distribution,
+    ~ alpha, #1
+    ~ beta, #0
+    "sex", # Parameters provided by researchers
+    "beta",
```

Comment:
I think you loose the `row tibble` advantage when you break the rows like this :-)